smalltalksrc/VMMaker/StackInterpreter.class.st

```diff
@@ -20741,15 +20741,8 @@ StackInterpreter >> stObject: array at: index put: value [
 	fmt := objectMemory formatOfHeader: hdr.
 	totalLength := objectMemory lengthOf: array baseHeader: hdr format: fmt.
 	fixedFields := objectMemory fixedFieldsOf: array format: fmt length: totalLength.
-	(fmt = objectMemory indexablePointersFormat
-	 and: [objectMemory isContextHeader: hdr])
-		ifTrue:
-			[stSize := self stackPointerForMaybeMarriedContext: array.
-			((self oop: index isGreaterThanOrEqualTo: 1)
-			 and: [(self oop: index isLessThanOrEqualTo: stSize)
-			 and: [self isStillMarriedContext: array]]) ifTrue:
-				[^self noInlineTemporary: index - 1 in: (self frameOfMarriedContext: array) put: value]]
-		ifFalse: [stSize := totalLength - fixedFields].
+	self deny: (objectMemory isContextHeader: hdr).
```

Comment:
This should instead fail the primitive (and add tests for it).

Like this, a mistaken use of the primitive will not cause a crash.